[BUG] Pandoc installation missing in documentation build page

### Describe the issue

While following the [Developer's guide for building documentation](https://www.aeon-toolkit.org/en/stable/developer_guide/documentation.html), after running the command `make html`, it gives the following error:

```
Notebook error:
PandocMissing in examples/anomaly_detection/anomaly_detection.ipynb:
Pandoc wasn't found.
Please check that pandoc is installed:
https://pandoc.org/installing.html
make: *** [Makefile:20: html] Error 2
```

Which suggests [Pandoc](https://pandoc.org/) is missing

### Suggest a potential alternative/fix

Simply [installing pandoc](https://pandoc.org/installing.html) worked on my system.

A simple note regarding this in the [Developer's guide for building documentation](https://www.aeon-toolkit.org/en/stable/developer_guide/documentation.html) should work, but I believe there can be better solutions.

### Additional context

I would be happy to work on this issue.

So I tried following the installation and documentation build steps on a fresh installation of Ubuntu (latest stable) (on wsl2) and I got the same error:
```
Notebook error:
PandocMissing in examples/anomaly_detection/anomaly_detection.ipynb:
Pandoc wasn't found.
Please check that pandoc is installed:
https://pandoc.org/installing.html
make: *** [Makefile:20: html] Error 2
```

I ran the following commands (in order):
```bash
python3 -m venv venv
```
```bash
pip install --editable .[dev]
```
```bash
pip install --editable .[docs]
```
```bash
cd docs
```
```bash
make html
```

After doing a bit of research, it turns out `Pandoc` can be installed via `apt` as well as via `pip`
So I ran the command `pip install pandoc` and then I ran `make html` and the build was still throwing the same error.

Finally, I tried installing `pandoc` via `apt`:
```
sudo apt install pandoc
```
This resulted in a successful build, and I was able to view all the build files.

I tried following the installation too, and it exited with the same error message: 
```sh
Notebook error:
PandocMissing in examples/anomaly_detection/anomaly_detection.ipynb:
Pandoc wasn't found.
Please check that pandoc is installed:
https://pandoc.org/installing.html
make: *** [html] Error 2
```

System info: Mac M1 Air (MacOS Sonoma)

--- a/docs/developer_guide/documentation.md
+++ b/docs/developer_guide/documentation.md
@@ -187,6 +187,21 @@ pip install --editable .[docs]
 cd docs
 ```
 
+> **Note:** If you are using Linux or MacOS, you'll have to install [**Pandoc**](https://pandoc.org/) as a dependency. Install it using the following command:
+>
+> If you're using any Debian based system, you can use the following command to install Pandoc:
+> ```bash
+> sudo apt install pandoc
+> ```
+>
+> If you're using MacOS, you can use the following command to install Pandoc:
+> ```bash
+> brew install pandoc
+> ```
+>
+> For all other operating systems, please checkout the [Pandoc installation guide](https://pandoc.org/installing.html).
+>
+
 3. To build the website locally, run:
 
 ```powershell

